app/demo-loop/Runner.cc

```diff
@@ -361,13 +353,15 @@ void Runner::build_primaries(RunnerInput const& inp)
  */
 void Runner::build_step_collectors(RunnerInput const& inp)
 {
-    std::shared_ptr<RootFileManager> root_manager_;
-    std::shared_ptr<StepCollector> step_collector_;
     StepCollector::VecInterface step_interfaces;
     if (!inp.mctruth_filename.empty())
     {
+        // Initialize ROOT file
         root_manager_
             = std::make_shared<RootFileManager>(inp.mctruth_filename.c_str());
+        write_to_root(inp, root_manager_.get());
+        // Must be called after all actions were added to action_registry
```

Comment:
This is why I put the ROOT writing at the end of the constructor. Can you move it back there? I think deleting the stray `root_manager_` local function will fix this all.